integration/spark2/src/main/spark2.2/CarbonSessionState.scala

```diff
@@ -256,6 +292,57 @@ class CarbonSqlAstBuilder(conf: SQLConf, parser: CarbonSpark2SqlParser) extends
     }
   }
 
+  override def visitChangeColumn(ctx: ChangeColumnContext): LogicalPlan = {
```

Comment:
Better you can catch the command at strategy rather than changing in parser

Reply:
We can catch the command in strategy. But as AlterTableAddColumnsCommand is only added in Spark 2.2. As our strategy is common for spark2.2. and spark2.1 it will be difficult to match the case of AlterTableAddColumnsCommand. Currently the SparkSqlAstBuilder is separated in 2.1 and 2.2 in CarbonSessionState.scala so it was easier to place it over there.